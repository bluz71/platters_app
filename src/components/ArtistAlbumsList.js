import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';
import '../styles/ArtistAlbumsList.css';

const renderTracks = (tracks) =>
  tracks.map((track, index) => <li key={index}>{track} </li>);

const renderAlbums = (albums, artistSlug, onYear, onGenre) =>
  albums.map((album) => (
    <div key={album.id} className="album">
      <Link to={`/artist/${artistSlug}/album/${album.slug}`}>
        <img
          className="img-responsive"
          src={album.cover_url}
          alt={album.title}
        />
      </Link>
      <div className="title">
        <Link to={`/artist/${artistSlug}/album/${album.slug}`}>
          <h2>{album.title}</h2>
        </Link>
        <span className="icon">
          <span className="filter" onClick={() => onYear(album.year)}>
            <FontAwesome name="calendar" /> {album.year}
          </span>
          <span className="filter" onClick={() => onGenre(album.genre)}>
            <FontAwesome name="tag" className="spacer-left-xsm" /> {album.genre}
          </span>
          <Link
            to={{
              pathname: `/artist/${artistSlug}/album/${album.slug}`,
              state: { scrollToComments: true }
            }}
          >
            <FontAwesome name="comment-o" className="spacer-left-xsm" />{' '}
            {album.comments_count}
          </Link>
        </span>
      </div>
      <h3 className="album-details">
        <small>
          {`(${album.tracks_count} Tracks, Time ${album.total_duration})`}
        </small>
      </h3>
      <ul>{renderTracks(album.tracks_summary)}</ul>
    </div>
  ));

const ArtistAlbumsList = ({ albums, artistSlug, onYear, onGenre }) => {
  return (
    <div className="ArtistAlbums">
      {renderAlbums(albums, artistSlug, onYear, onGenre)}
    </div>
  );
};

ArtistAlbumsList.propTypes = {
  albums: PropTypes.array,
  artistSlug: PropTypes.string,
  onYear: PropTypes.func,
  onGenre: PropTypes.func
};

export default ArtistAlbumsList;
