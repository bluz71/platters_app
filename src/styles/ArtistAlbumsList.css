.ArtistAlbums .album {
  border-bottom: 1px solid #eee;
}

.ArtistAlbums .album img {
  height: 80px;
  width: 80px;
  float: left;
  border: 1px solid #777;
  border-radius: 50%;
  margin-right: 1em;
}

.ArtistAlbums .album img:hover {
  border: 1px solid #3c9a5f;
}

.ArtistAlbums .album .title {
  height: 2.2em;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 1em 0 0;
}

.ArtistAlbums .album .title h2 {
  display: inline;
  font-size: 1.8em;
}

.ArtistAlbums .album .title a h2 {
  color: #515151;
  text-decoration: none;
}

.ArtistAlbums .album .title a h2:hover {
  color: #3c9a5f;
}

@media screen and (max-width: 992px) {
  .ArtistAlbums .album .title .icon {
    display: none;
  }
}

@media screen and (min-width: 992px) {
  .ArtistAlbums .album .title .icon {
    margin-top: 1em;
    float: right;
    font-size: 0.8em;
  }

  .ArtistAlbums .album .title a,
  .filter {
    cursor: pointer;
    color: #515151;
  }

  .ArtistAlbums .album .title a:hover,
  .filter:hover {
    color: #3c9a5f;
  }
}

.ArtistAlbums .album .album-details {
  margin-top: 0;
}

.ArtistAlbums .album .album-details small {
  font-size: 0.45em;
}
.ArtistAlbums .album ul {
  font-size: 0.8em;
  color: #515151;
  padding-left: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.ArtistAlbums .album ul li {
  display: inline;
  list-style-type: none;
}
