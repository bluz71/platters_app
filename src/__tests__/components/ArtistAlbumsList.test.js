import React from 'react';
import { shallow, mount } from 'enzyme';
import ArtistAlbumsList from '../../components/ArtistAlbumsList';

const albums = (args) => {
  return [
    {
      'id': 1,
      'title': 'ABC',
      'tracks_count': 2,
      'comments_count': 40,
      'total_duration': '15:31',
      'slug': 'abc',
      'cover_url': 'http://localhost:3000/small_XYZ--ABC.jpg',
      'year': 2008,
      'genre': 'Rock',
      'tracks_summary': [
        '1. First',
        '2. Second'
      ]
    },
    {
      'id': 2,
      'title': 'DEF',
      'tracks_count': 3,
      'comments_count': 26,
      'total_duration': '14:54',
      'slug': 'def',
      'cover_url': 'http://localhost:3000//small_XYZ--DEF.jpg',
      'year': 2014,
      'genre': 'Rock',
      'tracks_summary': [
        '1. First',
        '2. Second',
        '3. Third'
      ]
    }
  ];
};

describe('<ArtistAlbumsList />', () => {
  it('renders without crashing', () => {
    shallow(<ArtistAlbumsList albums={[]} />);
  });

  it('renders of a list of artist albums', () => {
    const wrapper = shallow(
      <ArtistAlbumsList albums={albums()} artistSlug="xyz" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
