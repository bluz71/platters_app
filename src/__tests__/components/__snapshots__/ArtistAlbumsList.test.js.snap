// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistAlbumsList /> renders of a list of artist albums 1`] = `
<div
  className="ArtistAlbums"
>
  <div
    className="album"
    key="1"
  >
    <Link
      replace={false}
      to="/artist/xyz/album/abc"
    >
      <img
        alt="ABC"
        className="img-responsive"
        src="http://localhost:3000/small_XYZ--ABC.jpg"
      />
    </Link>
    <div
      className="title"
    >
      <Link
        replace={false}
        to="/artist/xyz/album/abc"
      >
        <h2>
          ABC
        </h2>
      </Link>
      <span
        className="icon"
      >
        <a
          onClick={[Function]}
        >
          <FontAwesome
            name="calendar"
          />
           
          2008
        </a>
        <a
          onClick={[Function]}
        >
          <FontAwesome
            className="spacer-left-xsm"
            name="tag"
          />
           
          Rock
        </a>
        <Link
          replace={false}
          to={
            Object {
              "pathname": "/artist/xyz/album/abc",
              "state": Object {
                "scrollToComments": true,
              },
            }
          }
        >
          <FontAwesome
            className="spacer-left-xsm"
            name="comment-o"
          />
           
          40
        </Link>
      </span>
    </div>
    <h3
      className="album-details"
    >
      <small>
        (2 Tracks, Time 15:31)
      </small>
    </h3>
    <ul>
      <li
        key="0"
      >
        1. First
         
      </li>
      <li
        key="1"
      >
        2. Second
         
      </li>
    </ul>
  </div>
  <div
    className="album"
    key="2"
  >
    <Link
      replace={false}
      to="/artist/xyz/album/def"
    >
      <img
        alt="DEF"
        className="img-responsive"
        src="http://localhost:3000//small_XYZ--DEF.jpg"
      />
    </Link>
    <div
      className="title"
    >
      <Link
        replace={false}
        to="/artist/xyz/album/def"
      >
        <h2>
          DEF
        </h2>
      </Link>
      <span
        className="icon"
      >
        <a
          onClick={[Function]}
        >
          <FontAwesome
            name="calendar"
          />
           
          2014
        </a>
        <a
          onClick={[Function]}
        >
          <FontAwesome
            className="spacer-left-xsm"
            name="tag"
          />
           
          Rock
        </a>
        <Link
          replace={false}
          to={
            Object {
              "pathname": "/artist/xyz/album/def",
              "state": Object {
                "scrollToComments": true,
              },
            }
          }
        >
          <FontAwesome
            className="spacer-left-xsm"
            name="comment-o"
          />
           
          26
        </Link>
      </span>
    </div>
    <h3
      className="album-details"
    >
      <small>
        (3 Tracks, Time 14:54)
      </small>
    </h3>
    <ul>
      <li
        key="0"
      >
        1. First
         
      </li>
      <li
        key="1"
      >
        2. Second
         
      </li>
      <li
        key="2"
      >
        3. Third
         
      </li>
    </ul>
  </div>
</div>
`;
